CodeEditModules/Modules/Git/src/Client/Models/ChangedFiles.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  ChangedFiles.swift
+//  
+//
+//  Created by Nanashi Li on 2022/05/05.
+//
+
+import Foundation
+import SwiftUI
+import WorkspaceClient
+
+public struct ChangedFiles: Codable, Hashable, Identifiable {
+    public var id = UUID()
+    public var changeType: String
+    public var fileLink: String
+
+    /// Use it like this
+    /// ```swift
+    /// Image(systemName: item.systemImage)
+    /// ```
+    public var systemImage: String {
+        if !isFile {
+            return "folder.fill"
+        } else {
+            return FileIcon.fileIcon(fileType: fileType)
+        }
+    }
+
+    /// Returns the file name (e.g.: `Package.swift`)
+    public var fileName: String {
+        URL(string: fileLink)?.lastPathComponent ?? ""
+    }
+
+    private var isFile: Bool {
+        return fileName.contains(".")
+    }
+
+    /// Returns the extension of the file or an empty string if no extension is present.
+    private var fileType: String {
+        URL(string: fileLink)?.lastPathComponent.components(separatedBy: ".").last ?? ""
```

Comment:
the _URL_ has an extension API: https://developer.apple.com/documentation/foundation/url/1780487-pathextension

Reply:
Again multiple files will need to be changed. I will only change mine.

Reply:
this could indeed be a problem if the file is file.extension?query=parameter

Reply:
@nanashili please address the comment, there is a simple API that Apple offers and as @wdg mentioned the logic above it's wrong.